src/crypto/public_key/elliptic/eddsa.js

```diff
@@ -69,4 +69,51 @@ async function verify(oid, hash_algo, { R, S }, m, publicKey, hashed) {
   return nacl.sign.detached.verify(hashed, signature, publicKey.subarray(1));
 }
 
-export default { sign, verify };
+/**
+ * Validate EdDSA parameters
+ * @param {module:type/oid}    oid Elliptic curve object identifier
+ * @param {Uint8Array}         Q   EdDSA public point
+ * @param {Uint8Array}         k   EdDSA secret seed
+ * @returns {Promise<Boolean>} whether params are valid
+ * @async
+ */
+async function validateParams(oid, Q, k) {
+  // Check whether the given curve is supported
+  if (oid.getName() !== 'ed25519') {
+    return false;
+  }
+
+  /**
+   * Derive public point Q' = dG from private key
+   * and expect Q == Q'
+   */
+  const { publicKey } = nacl.sign.keyPair.fromSeed(k);
+  const dG = [0x40, ...publicKey]; // Add public key prefix
+  return util.equalsArray(Q, dG);
+}
+
+/**
+ * Parses MPI params and returns them as byte arrays of fixed length
+ * @param {Array} params key parameters
+ * @returns {Object} parameters in the form
+ *  { oid, seed: Uint8Array, Q: Uint8Array }
+ */
+function parseParams(params) {
+  if (params.length < 2 || params.length > 3) {
+    throw new Error('Unexpected number of parameters');
+  }
+
+  const parsedParams = {
+    oid: params[0],
+    Q: params[1].toUint8Array('be', 33)
+  };
+
+  if (params.length === 3) {
+    parsedParams.seed = params[2].toUint8Array('be', 32);
+  }
+
+  return parsedParams;
+}
```

Comment:
I'm still not really a big fan of these `parseParams` functions, sorry. First of all, they don't really parse anything :see_no_evil: Basically all they do is give the params a name, and say how long they should be. We already have:
https://github.com/openpgpjs/openpgpjs/blob/e411f8e3fd40a79555a94b8cc70d89df76247916/src/crypto/crypto.js#L193-L201
In the future, we could change that to, for example,
```js
        return { n: type_mpi, e: type_mpi };
```
Then, we can change `constructParams` to return an object similar to the one you're creating here.
Then we still have to think about the length, but maybe we can also encode that there somehow.
Now - I don't think we have to do all of that in this PR, but I do worry that some of these changes would involve reverting the changes in this PR.
So - even though it sounds stupid, I would tend to just keep the code as it was (calling `Uint8Array` in the places where the parameters are used). It will cause some code duplication, but it will make it easier to make the above changes during parsing, instead of during usage.

Reply:
I think we use `constructParams` in a way that is too generic & low level to be converted so that it has enough context to include the lengths (which depend on some of the parameters we are parsing).
In general, it seems to me that wherever we use `constructParams`, we also know the cipher algo in use, which means we can call a cipher-specific function to parse/normalize the params ... Leaving aside the name, `parseParams` can take the array that `constructParams` returns and process it accordingly. 
IMO, what we could do to take the best of both functions is to extend the MPI class to include an optional `length` field,  then `parseParams` can add the `length` info to the MPIs returned by `constructParams` so that calling `MPI.toUint8Array()` already results in a properly-sized array.

(but yes, this ^ proposed change would still require reverting the changes here)

Reply:
> In general, it seems to me that wherever we use constructParams, we also know the cipher algo in use, which means we can call a cipher-specific function to parse/normalize the params

Good point. Even where we don't (at least at compile time), we could have a high-level `parseParams` function with a switch statement on the algorithm ID which calls the cipher-specific function. We can then even do away with `getPubKeyParamTypes`, and `parseParams` could call `mpi.read` to directly parse from binary into `{ n, e }` object, etc.

> IMO, what we could do to take the best of both functions is to extend the MPI class to include an optional length field, then parseParams can add the length info to the MPIs returned by constructParams so that calling MPI.toUint8Array() already results in a properly-sized array.

I would prefer to just return a Uint8Array directly, and make that the "canonical" representation of this information; if some code needs an MPI or BN, we can do `new BN()` there.

But yeah, we can do all of that later, and merge this PR as is, as long as we clean things up later :)